src/main/java/org/prebid/server/bidder/sspbc/SspbcBidder.java

```diff
@@ -0,0 +1,315 @@
+        }
+
+        final URIBuilder uriBuilder = getUri(site.getPage());
+        siteBuilder.domain(uriBuilder.getHost());
+
+        return siteBuilder.build();
+    }
+
+    private static Integer updateTest(Integer requestType, Integer test) {
+        return Objects.equals(requestType, REQUEST_TYPE_TEST) ? 1 : test;
+    }
+
+    private String getImpSize(Imp imp) {
+        if (imp.getBanner() == null || imp.getBanner().getFormat().size() == 0) {
+            return IMP_FALLBACK_SIZE;
+        }
+
+        long areaMax = 0L;
+        String impSize = IMP_FALLBACK_SIZE;
+
+        for (Format format : imp.getBanner().getFormat()) {
+            int area = format.getW() * format.getH();
+            if (area > areaMax) {
+                areaMax = area;
+                impSize = String.format("%dx%d", format.getW(), format.getH());
+            }
+        }
+
+        return impSize;
+    }
+
+    private Integer getRequestType(BidRequest request) {
+        int incompleteImps = 0;
+
+        for (Imp imp : request.getImp()) {
+            final ExtImpSspbc extImpSspbc = parseImpExt(imp);
+            if (extImpSspbc.getTest() != 0) {
+                return REQUEST_TYPE_TEST;
+            }
+
+            if (StringUtils.isEmpty(extImpSspbc.getSiteId()) || StringUtils.isEmpty(extImpSspbc.getId())) {
+                incompleteImps += 1;
+            }
+        }
+
+        if (incompleteImps > 0) {
+            return REQUEST_TYPE_ONE_CODE;
+        }
+
+        return REQUEST_TYPE_STANDARD;
+    }
+
+    private ExtImpSspbc parseImpExt(Imp imp) {
+        try {
+            return mapper.mapper().convertValue(imp.getExt(), SSPBC_EXT_TYPE_REFERENCE).getBidder();
+        } catch (IllegalArgumentException e) {
+            throw new PreBidException(e.getMessage());
+        }
+    }
+
+    private HttpRequest<BidRequest> createRequest(BidRequest request) {
+        return HttpRequest.<BidRequest>builder()
+                .method(HttpMethod.POST)
+                .uri(updateUrl(getUri(endpointUrl)))
+                .headers(HttpUtil.headers())
+                .body(mapper.encodeToBytes(request))
+                .payload(request)
+                .build();
+    }
+
+    private URIBuilder getUri(String endpointUrl) {
+        final URI uri;
+        try {
+            uri = new URI(endpointUrl);
+        } catch (URISyntaxException e) {
+            throw new PreBidException("Malformed URL: %s.".formatted(endpointUrl));
+        }
+        return new URIBuilder(uri);
+    }
+
+    private String updateUrl(URIBuilder uriBuilder) {
+        return uriBuilder
+                .addParameter("bdver", ADAPTER_VERSION)
+                .addParameter("inver", PREBID_SERVER_INTEGRATION_TYPE)
+                .toString();
+    }
+
+    @Override
+    public Result<List<BidderBid>> makeBids(BidderCall<BidRequest> httpCall, BidRequest bidRequest) {
+        final List<BidderError> errors = new ArrayList<>();
```

Comment:
Why do we need this list if implementation of `makeBids` returns an error once it occurs?